core/server/worker/src/main/java/alluxio/worker/block/io/DeStoreBlockReader.java

```diff
@@ -0,0 +1,62 @@
+package alluxio.worker.block.io;
+
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import io.netty.buffer.ByteBuf;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * An reader class with metrics.
+ */
+public class DeStoreBlockReader extends BlockReader {
+  private final BlockReader mDeBlockReader;
+
+  /**
+   * A decorator of BlockReader.
+   * @param deblockReader block reader
+   */
+  public DeStoreBlockReader(BlockReader deblockReader) {
```

Comment:
Requiring the reader to be a `BlockReader` is too general: `UnderFileSystemBlockReader` is also a `BlockReader`, but it's obviously wrong to use that with the metric `MetricKey.WORKER_BYTES_READ_CACHE`.

Since we are dealing with the bytes-read-cache metric here, requiring the argument to be of type `LocalFileBlockReader` makes sense.